flow/project.py

```diff
@@ -1821,39 +1959,66 @@ def _fetch_status(self, jobs, err, ignore_errors, status_parallelization='thread
         self._fetch_scheduler_status(jobs, err, ignore_errors)
         # Get status dict for all selected jobs
 
-        def _print_progress(x):
-            print("Updating status: ", end='', file=err)
-            err.flush()
-            n = max(1, int(len(jobs) / 10))
-            for i, _ in enumerate(x):
-                if (i % n) == 0:
-                    print('.', end='', file=err)
-                    err.flush()
-                yield _
+        def _print_status(iterable, fetch_status, description):
+            t = time.time()
+            num_itr = len(iterable)
+            results = []
+            for i, itr in enumerate(iterable):
+                results.append(fetch_status(itr))
+                if time.time() - t > 0.2:  # status interval
```

Comment:
why 0.2 (seconds?)

Reply:
This was there before I started editing it, I just refactored it a little bit.
Maybe @csadorf or @b-butler could help explain this

Reply:
I must have missed that in the diff, 0.2 sounds reasonable, I was just curious how that value was chosen. Not a big deal but could be nice to add as a code comment.

Reply:
I don't remember how this number was chosen, obviously it's a heuristic but I don't recall the rationale. I would suggest looking at #110, that's where it was added iirc.

Reply:
Thank you @vyasr 
@mikemhenry I'll add a code comment.